File: 2. Structural Design Patterns/Facade/Typescript/FacadeTemplate.ts
// Facade Pattern – TypeScript Template

class SubsystemA { opA(){ /* ... */ } }
class SubsystemB { opB(){ /* ... */ } }
class SubsystemC { opC(){ /* ... */ } }

class Facade { private a = new SubsystemA(); private b = new SubsystemB(); private c = new SubsystemC(); doWork(){ this.a.opA(); this.b.opB(); this.c.opC(); } }

(function main(){ new Facade().doWork() })()

export {}

File: 2. Structural Design Patterns/Facade/C#/FacadeTemplate.cs
// Facade Pattern – C# Template

namespace FacadeTemplate
{
    class SubsystemA { public void OpA(){ /* ... */ } }
    class SubsystemB { public void OpB(){ /* ... */ } }
    class SubsystemC { public void OpC(){ /* ... */ } }

    class Facade
    {
        private readonly SubsystemA a = new();
        private readonly SubsystemB b = new();
        private readonly SubsystemC c = new();
        public void DoWork(){ a.OpA(); b.OpB(); c.OpC(); }
    }

    class Program { static void Main(){ var f = new Facade(); f.DoWork(); } }
}
